pkg/cephfs/volume.go

```diff
@@ -17,140 +17,109 @@ limitations under the License.
 package cephfs
 
 import (
-	"fmt"
-	"os"
-	"path"
+	"strconv"
+	"strings"
+
+	"github.com/ceph/ceph-csi/pkg/util"
 
 	"k8s.io/klog"
 )
 
 const (
-	cephVolumesRoot = "csi-volumes"
+	namespacePrefix = "fsvolumens_"
+)
 
-	namespacePrefix = "ns-"
+var (
+	cephfsInit = false
 )
 
-func getCephRootPathLocal(volID volumeID) string {
-	return fmt.Sprintf("%s/controller/volumes/root-%s", PluginFolder, string(volID))
-}
+func getVolumeRootPathCeph(volOptions *volumeOptions, cr *credentials, volID volumeID) string {
+	stdout, _, err := util.ExecCommand(
+		"ceph",
+		"fs",
+		"subvolume",
+		"getpath",
```

Comment:
I was expecting a `getbasepath` command, rather than a `getpath` command, such that we need to call it only once and can construct the subvolume name as `basepath` + '/' + `volID`. It would optimize a call to Ceph to retrieve this data for each Create/Delete/Mount as needed. @ajarr something to think about for the future, or not a viable alternative?

Reply:
Yes, we can do this, `ceph fs subvolumegroup getpath`. Added a tracker ticket,
http://tracker.ceph.com/issues/40617

Reply:
I don't like the idea of Ceph CSI constructing the `subvolume path` based on `subvolumegroup path`.  Aren't we breaking the abstraction if we let CSI construct subvolume path based on the subvolumegroup path?
@ShyamsundarR

Reply:
The way I look at it the `VolID` that we pass is the `subvolume` name, hence `subvolbasepath` + `subvolname` (that we set) is the subvolume path. As a result I do not see this as an abstraction break, but maybe I am reading it this way because I want to :)

My intention for the base path is to avoid repeated calls to Ceph to fetch the same, as a result improving per operation performance by that much.